### PR TITLE
add right management

### PR DESCRIPTION
**Pull request by orthagh**
See: http://glpi-developer-documentation.readthedocs.io/en/acl/

### CODE REVIEW

File: source/devapi/acl.rst [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+
+Provide a way for administrator to segment usages into profiles of users.
+
+
+Profiles
+^^^^^^^^
+
+The `Profile class <https://forge.glpi-project.org/apidoc/class-Profile.html>`_ (corresponding to ``glpi_profiles`` table) stores each set of rights.
+
+A profile have a set of base fields independent of sub rights and, so, could:
```
Comment (btry) [inline source/devapi/acl.rst]: "a profile has " instead of "a profile have" ?

File: source/devapi/acl.rst [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+- a profile foreign key (``profiles_id`` field)
+- a key (``name`` field)
+- a value (``right`` field)
+
+The keys match the static property ``$rightname`` in the GLPI itemtypes.
+Ex: In Computer class, we have a ``static $rightname = 'computer';``
+
+Value is a numeric sum of integer constants.
+
+Bases right values can be found in inc/define.php:
```
Comment (trasher) [inline source/devapi/acl.rst]: "Bases" Sounds strange to me

File: source/devapi/acl.rst [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+   define("ALLSTANDARDRIGHT", 31);
+   define("READNOTE", 32);
+   define("UPDATENOTE", 64);
+   define("UNLOCK", 128);
+
+So, for example, to have the right to READ and UPDATE an itemtype, we'll have a ``right`` value of 3.
+
+As defined in this above block, we have a computation of all standards right = 31:
+
+READ (1)
```
Comment (trasher) [inline source/devapi/acl.rst]: This is displayed inline. Use a litteral block (http://docutils.sourceforge.net/docs/ref/rst/restructuredtext.html#literal-blocks) or a `code-block` to get this displayed as in the sources.

File: source/devapi/acl.rst [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+\+ UPDATE (2)
+\+ CREATE (4)
+\+ DELETE (8)
+\+ PURGE (16)
+= 31
+
+If you need to extends the possible values of rights, you need to declare theses part into your itemtype, simplified example for Ticket class:
```
Comment (trasher) [inline source/devapi/acl.rst]: these

File: source/devapi/acl.rst [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+
+         $values[self::READASSIGN] = array('short' => __('See assigned'),
+                                           'long'  => __('See assigned tickets'));
+
+         return $values;
+      }
+
+      ...
+
+The new rights need to be checked by your own functions, see checkrights__
```
Comment (trasher) [inline source/devapi/acl.rst]: I guess you want a link? Actually, this one points to nothing.

Comment (trasher) [inline source/devapi/acl.rst]: You can either use paragraph exact title (but you should avoid) or a reference.

File: source/devapi/acl.rst [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+
+
+Check rights
+^^^^^^^^^^^^
+
+Each itemtype class which inherits from CommonDBTM will benefit from standard right checks.
```
Comment (trasher) [inline source/devapi/acl.rst]: Maybe add a link to the `COmmonDBTM` part of the doc?

File: source/devapi/acl.rst [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+* `haveRightsAnd <https://forge.glpi-project.org/apidoc/class-Session.html#_haveRightsAnd>`_
+* `haveRightsOr <https://forge.glpi-project.org/apidoc/class-Session.html#_haveRightsOr>`_
+
+All above functions return a boolean. If we want a graceful die of the page we have equivalent function but with a ``check`` prefix instead ``have``:
+
+* `checkRight <https://forge.glpi-project.org/apidoc/class-Session.html#_checkRight>`_
+* `checkRightsAnd <https://forge.glpi-project.org/apidoc/class-Session.html#_checkRightsAnd>`_
+* `checkRightsOr <https://forge.glpi-project.org/apidoc/class-Session.html#_checkRightsOr>`_
+
+If you need to check a right directly in a SQL query, use a logical ``&`` and ``|`` operators, ex for users:
```
Comment (trasher) [inline source/devapi/acl.rst]: Add a link to the documentation for those operators

File: source/devapi/acl.rst [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+   abstract class CommonDBRelation extends CommonDBConnexity {
+      static public $checkItem_1_Rights = self::HAVE_SAME_RIGHT_ON_ITEM;
+      static public $checkItem_2_Rights = self::HAVE_SAME_RIGHT_ON_ITEM;
+
+      ...
+   }
+
+possible values for theses properties are:
+
+* ``DONT_CHECK_ITEM_RIGHTS``:  Don't check the parent, we always have all rights regardless of parent's rights.
```
Comment (trasher) [inline source/devapi/acl.rst]: s/Don/don/ (no caps after `:`)

File: source/devapi/acl.rst [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+   }
+
+   abstract class CommonDBRelation extends CommonDBConnexity {
+      static public $checkItem_1_Rights = self::HAVE_SAME_RIGHT_ON_ITEM;
+      static public $checkItem_2_Rights = self::HAVE_SAME_RIGHT_ON_ITEM;
+
+      ...
+   }
+
+possible values for theses properties are:
```
Comment (trasher) [inline source/devapi/acl.rst]: these

**Pull request merged**